scripts/monitoring/cron-send-logging-checks.py

```diff
@@ -0,0 +1,227 @@
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+#This is not a module, but pylint thinks it is.  This is a command.
+#pylint: disable=invalid-name
+#If a check throws an exception it is failed and should alert
+#pylint: disable=bare-except
+import argparse
+import ssl
+import urllib2
+
+# pylint: disable=import-error
+from openshift_tools.monitoring.ocutil import OCUtil
+from openshift_tools.monitoring.zagg_sender import ZaggSender
+# pylint: enable=import-error
+
+class OpenshiftLoggingStatus(object):
+    '''
+        This is a check for the entire EFK stack shipped with OCP
+    '''
+    def __init__(self):
+        ''' Initialize OpenShiftLoggingStatus class '''
+        self.zagg_sender = None
+        self.oc = None
+        self.args = None
+        self.es_pods = []
+        self.fluentd_pods = []
+
+        es_cert = '/etc/elasticsearch/secret/admin-'
+        self.es_curl = "curl --cert {}cert --key {}key --cacert {}ca -XGET ".format(es_cert, es_cert, es_cert)
+
+    def parse_args(self):
+        ''' Parse arguments passed to the script '''
+        parser = argparse.ArgumentParser(description='OpenShift Cluster Logging Checker')
+        parser.add_argument('-v', '--verbose', action='store_true', default=None, help='Verbose output')
+        parser.add_argument('--debug', action='store_true', default=None, help='Debug?')
+
+        self.args = parser.parse_args()
+
+    def get_pods(self):
+        ''' Get all pods and filter them in one pass '''
+        pods = self.oc.get_pods()
+        for pod in pods['items']:
+            if 'component' in pod['metadata']['labels']:
+                # Get ES pods
+                if pod['metadata']['labels']['component'] == 'es':
+                    self.es_pods.append(pod)
+                elif pod['metadata']['labels']['component'] == 'fluentd':
+                    self.fluentd_pods.append(pod)
+
+    def check_elasticsearch(self):
+        ''' Various checks for elasticsearch '''
+        es_status = {}
+        es_status['single_master'] = None
+        es_master_name = None
+        es_status['pods'] = {}
+        for pod in self.es_pods:
+            pod_dc = pod['metadata']['labels']['deploymentconfig']
+            pod_name = pod['metadata']['name']
+            es_status['pods'][pod_dc] = {}
+
+
+            # exec into a pod and get cluster health
+            curl_cmd = "{} 'https://localhost:9200/_cluster/health?pretty=true'".format(self.es_curl)
```

Comment:
What happens if the oc exec succeeds, but the curl itself fails?

Like if elastic search is in a bad state or down for whatever reason?

From my cursory view, that seems like it would cause an ops-runner alert (as this script would back trace).  ops-runner should only alert if there's a bug in the script. The script itself should report typical failure cases.